client/securedrop_client/gui/widgets.py

```diff
@@ -639,16 +793,29 @@ def __init__(
         # Create widgets
         self.view_holder = QWidget()
         self.view_holder.setObjectName("MainView_view_holder")
-        self.view_layout = QVBoxLayout()
+
+        # Layout where only one view shows at a time. Suitable for the case
+        # where we show either a conversation or a contextually-appropriate
+        # message ("Select a source...", "Nothing to see yet", etc)
+        self.view_layout = QStackedLayout()
```

Comment:
This is a change worth calling out: instead of using one Vertical layout and hiding/showing child widgets within it (and then having to test like `assert layout.child_one.isHidden()` and manage state of a bunch of individual widgets), [QStackedLayout](https://doc.qt.io/qt-5/qstackedlayout.html) is a better option.

A further/bonus perk, which I have not included because of scope creep, is that with a StackedLayout, we could decide to cache the n most recent expanded source conversation pages, and push/pop from the stack, instead of what we do now, which is maintain a non-size-constrained in memory dictionary of widget elements.